common/views/components/styled/ResponsiveTable.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+import styled from 'styled-components';
+import type { ComponentType } from 'react';
+
+export type Props = {
+  headings: string[],
+};
+
+const StyledTable: ComponentType<Props> = styled.table`
+  width: 100%;
+  border-collapse: collapse;
+
+  /* Force table to not be like tables anymore */
+  table,
+  thead,
+  tbody,
+  th,
+  td,
+  tr {
+    display: block;
+    width: 100%;
+  }
+
+  /* Hide table headers (but not display: none;, for accessibility) */
+  thead tr {
+    position: absolute;
+    top: -9999px;
+    left: -9999px;
+  }
+
+  tr {
+    margin-bottom: 24px;
+  }
+
+  th,
+  td {
+    min-width: 150px;
+    text-align: left;
+    font-weight: normal;
+    padding-right: 30px;
+  }
+
+  td {
+    /* Behave  like a "row" */
+    border: none;
+    position: relative;
+    padding-left: 50%;
+    margin-bottom: 6px;
+  }
+
+  td:before {
+    /* Now like a table header */
+    position: absolute;
+    /* Top/left values mimic padding */
+    top: 0;
+    left: 0;
+    width: 45%;
+    white-space: nowrap;
+    speak: none;
+  }
+
+  ${props =>
+    props.headings.map(
+      (heading, i) =>
+        `td:nth-of-type(${i + 1}):before {
+          content: '${heading}';
+        }`
+    )}
```

Comment:
Is this accessible? Or should we just be building the table via HTML?

Reply:
I think it should be fine, the actual table headers are still available to screen readers and replacing them visually on small screen devices for each row I think makes it easier to read